### DISCUSSION TITLE
[HttpKernel 7.0]Symfony\Component\HttpKernel\Bundle\Bundle::$path must not be accessed before initialization

### DISCUSSION THREAD

**Discussion by tacman** [symfony/symfony#52292] [Q&A]
I'm testing a bundle that works in 6.4 with Symfony 7 and running into an error. 

I don't know if this is a bug in HttpKernel or if something has changes when extending AbstractBundle, so I'm reporting it.

My bundle initialization is pretty basic.

https://github.com/survos/SurvosCrawlerBundle/blob/main/src/SurvosCrawlerBundle.php

```php
namespace Survos\CrawlerBundle;

use Survos\CrawlerBundle\Command\CrawlCommand;
use Survos\CrawlerBundle\Controller\CrawlerController;
use Survos\CrawlerBundle\Services\CrawlerService;
use Survos\CrawlerBundle\Twig\TwigExtension;
use Symfony\Component\Config\Definition\Configurator\DefinitionConfigurator;
use Symfony\Component\DependencyInjection\ContainerBuilder;
use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
use Symfony\Component\DependencyInjection\Reference;
use Symfony\Component\HttpKernel\Bundle\AbstractBundle;
use Symfony\Component\HttpKernel\Bundle\Bundle;
use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;

class SurvosCrawlerBundle extends AbstractBundle
{
    /**
     * @param array<mixed> $config
     */
    public function loadExtension(array $config, ContainerConfigurator $container, ContainerBuilder $builder): void
    {
        // config
    }

    public function configure(DefinitionConfigurator $definition): void
    {
        ;
    }
}
```

Example:

```bash
symfony new command-7 --webapp --version=next --php=8.2 && cd command-7
composer config extra.symfony.allow-contrib true
sed -i 's/"6.4.*"/"^7.0"/' composer.json
composer update
composer req survos/command-bundle
```

Symfony operations: 1 recipe (1690234b1ee9712d097f9393129adc58)
  - Configuring survos/command-bundle (>=1.5): From github.com/symfony/recipes-contrib:main
Executing script cache:clear [KO]
 [KO]
Script cache:clear returned with error code 255
!!  Error {#111
!!    #message: "Typed property Symfony\Component\HttpKernel\Bundle\Bundle::$path must not be accessed before initialization"
!!    #code: 0
!!    #file: "./vendor/symfony/http-kernel/Bundle/AbstractBundle.php"
!!    #line: 53
!!    trace: {
!!      ./vendor/symfony/http-kernel/Bundle/AbstractBundle.php:53 { …}
!!      ./vendor/symfony/http-kernel/Kernel.php:546 { …}
!!      ./vendor/symfony/http-kernel/Kernel.php:633 { …}
!!      ./vendor/symfony/http-kernel/Kernel.php:587 { …}
!!      ./vendor/symfony/http-kernel/Kernel.php:487 { …}
!!      ./vendor/symfony/http-kernel/Kernel.php:723 { …}
!!      ./vendor/symfony/http-kernel/Kernel.php:123 { …}
!!      ./vendor/symfony/framework-bundle/Console/Application.php:173 { …}
!!      ./vendor/symfony/framework-bundle/Console/Application.php:71 { …}
!!      ./vendor/symfony/console/Application.php:169 { …}
!!      ./vendor/symfony/runtime/Runner/Symfony/ConsoleApplicationRunner.php:49 { …}
!!      ./vendor/autoload_runtime.php:29 { …}
!!      ./bin/console:11 {
!!        › 
!!        › require_once dirname(__DIR__).'/vendor/autoload_runtime.php';
!!        › 
!!        arguments: {
!!          "/home/tac/trash/command-7/vendor/autoload_runtime.php"
!!        }
!!      }
!!    }
!!  }
!!  2023-10-25T13:41:44+00:00 [critical] Uncaught Error: Typed property Symfony\Component\HttpKernel\Bundle\Bundle::$path must not be accessed before initialization
!!

**Comment by ghost** [comment]
Yes, this is a bug in AbstractBundle. Property `Bundle::$path` was typed with string in 7.0 but the method `AbstractBundle::getPath()` still does `if (null === $this->path)` while parent's method does `if (!isset($this->path))`.

Thank you for testing upcoming Symfony 7.0 version.

**Reply by tacman** [reply]
Okay if I make a PR that does that?

**Reply by ghost** [reply]
Yes. Just fork symfony/symfony and open a PR with a fix.
